helix-term/src/commands/lsp.rs

```diff
@@ -915,23 +915,41 @@ where
     F: Future<Output = helix_lsp::Result<serde_json::Value>> + 'static + Send,
 {
     let (view, doc) = current!(cx.editor);
+    let mut futures: FuturesOrdered<_> = doc
+        .language_servers_with_feature(feature)
+        .map(|language_server| {
+            let offset_encoding = language_server.offset_encoding();
+            let pos = doc.position(view.id, offset_encoding);
+            let future = request_provider(language_server, pos, doc.identifier()).unwrap();
 
-    let language_server = language_server_with_feature!(cx.editor, doc, feature);
-    let offset_encoding = language_server.offset_encoding();
-    let pos = doc.position(view.id, offset_encoding);
-    let future = request_provider(language_server, pos, doc.identifier()).unwrap();
+            async move {
+                let json = future.await?;
+                let response = serde_json::from_value::<Option<lsp::GotoDefinitionResponse>>(json)?;
+                anyhow::Ok((response, offset_encoding))
+            }
+        })
+        .collect();
 
-    cx.callback(
-        future,
-        move |editor, compositor, response: Option<lsp::GotoDefinitionResponse>| {
-            let items = to_locations(response);
+    cx.jobs.callback(async move {
+        let mut items: Vec<Location> = Vec::new();
+        let mut offset_encoding = OffsetEncoding::default();
+
+        while let Some((location, encoding)) = futures.try_next().await? {
+            items = to_locations(location);
+            offset_encoding = encoding;
```

Comment:
It looks like this uses the response from the last language server to respond successfully. I believe we want to merge the items instead

Reply:
My test case for this was configuring typescript files to have both the typescript and angular language servers configured, in that order. So usually the first response from the typescript language server was fine, except for template strings, where angular would respond instead. So simply using the first valid response from the configured language servers in order was sufficient. But I think this should be fine as well, I just didn't want for goto definition to open a selection popup in the case that multiple language servers respond.